numerics/fit_hermite_spline.hpp

```diff
@@ -0,0 +1,41 @@
+﻿#pragma once
+
+#include <list>
+
+#include "numerics/hermite3.hpp"
+
+namespace principia {
+namespace numerics {
+namespace internal_fit_hermite_spline {
+
+// Given |samples| for which the arguments, values, and derivatives can be
+// obtained via the given functors, returns a sequence of iterators
```

Comment:
You need to document the functors better, they are totally mysterious in the template declarations below.

In fact it would be much more readable to do something like, e.g.:
```
  typename Argument,
  typename Value,

  std::function<Argument(typename Samples::iterator)> const& get_argument,
  ... // Same for get_value.
  std::function<Derivative<Value>(typename Sample::iterator)> const& get_derivative,
```
As written, all the contract is hidden in the implementation.

Reply:
Done. https://twitter.com/stephentyrone/status/902227449433513984